File: apps/couple-diary/front/lib/models/post.dart
class Post {
  final int id;
  final int roomId;
  final int? authorId;
  final String? text;
  final List<String> photos;
  final String? mood;
  final DateTime? shotAt;
  final DateTime createdAt;

  Post({
    required this.id,
    required this.roomId,
    this.authorId,
    this.text,
    required this.photos,
    this.mood,
    this.shotAt,
    required this.createdAt,
  });

  factory Post.fromJson(Map<String, dynamic> j) => Post(
    id: j['id'],
    roomId: j['room_id'],
    authorId: j['author_id'],
    text: j['text'],
    photos: (j['photos'] as List).cast<String>(),
    mood: j['mood'],
    shotAt: j['shot_at'] != null ? DateTime.parse(j['shot_at']) : null,
    createdAt: DateTime.parse(j['created_at']),
  );
}

File: apps/couple-diary/server/app/schemas/post.py
from datetime import datetime

from pydantic import BaseModel


class PostIn(BaseModel):
    text: str | None = None
    photos: list[str] = []
    mood: str | None = None
    shot_at: datetime | None = None


class PostOut(BaseModel):
    id: int
    room_id: int
    author_id: int | None
    text: str | None
    photos: list[str]
    mood: str | None
    shot_at: datetime | None
    created_at: datetime

    class Config: from_attributes = True


class CommentIn(BaseModel):
    post_id: int
    text: str


class CommentOut(BaseModel):
    id: int
    post_id: int
    author_id: int | None
    text: str
    created_at: datetime

    class Config: from_attributes = True
